include/systems/inter_mesh_projection.h

```diff
@@ -37,71 +37,71 @@
-        // projected onto the _to_mesh
-        InterMeshProjection(System & _from_system, System & _to_mesh);
+  // Constructor, specifies the _from_system whose vectors will be
+  // projected onto the _to_mesh
+  InterMeshProjection(System & _from_system, System & _to_mesh);
 
-        // Projects from_system vectors onto the to_mesh
-        void project_system_vectors();
+  // Projects from_system vectors onto the to_mesh
+  void project_system_vectors();
 
-        static Number fptr(const Point & p, const Parameters &, const std::string & libmesh_dbg_var(sys_name), const std::string & unknown_name);
+  static Number fptr(const Point & p, const Parameters &, const std::string & libmesh_dbg_var(sys_name), const std::string & unknown_name);
 
-        static Gradient gptr(const Point & p, const Parameters &, const std::string & libmesh_dbg_var(sys_name), const std::string & unknown_name);
+  static Gradient gptr(const Point & p, const Parameters &, const std::string & libmesh_dbg_var(sys_name), const std::string & unknown_name);
 
-        private:
+private:
 
-        // Local copy of the _from_system
-        System & from_system;
+  // Local copy of the _from_system
+  System & from_system;
 
-        // Local copy of the _to_system
-        System & to_system;
+  // Local copy of the _to_system
+  System & to_system;
 
-    };
+};
 
-    // This class provides the functor we will supply to System::project_vector
-    // inside InterMeshProjection::project_system_vectors.
-    // Object is constructed by passing in a pointer to the mesh function whose
-    // gradient we want to shim via operator().
-    class GradientMeshFunction : public FunctionBase<Gradient>
-    {
-        public:
-        // Constructor
-        GradientMeshFunction(MeshFunction * _mesh_function);
+// This class provides the functor we will supply to System::project_vector
+// inside InterMeshProjection::project_system_vectors.
+// Object is constructed by passing in a pointer to the mesh function whose
+// gradient we want to shim via operator().
+class GradientMeshFunction : public FunctionBase<Gradient>
+{
+public:
+  // Constructor
+  GradientMeshFunction(MeshFunction * _mesh_function);
 
-        // Destructor
-        virtual ~GradientMeshFunction () { }
+  // Destructor
+  virtual ~GradientMeshFunction () { }
 
-        virtual void init () { }
+  virtual void init () { }
 
-        virtual std::unique_ptr<FunctionBase<Gradient>> clone () const
-        {
-          //return libmesh_make_unique<GradientMeshFunction>(dynamic_cast<MeshFunction *>(mesh_function.get()));
-          return libmesh_make_unique<GradientMeshFunction>(mesh_function.get());
-        }
+  virtual std::unique_ptr<FunctionBase<Gradient>> clone () const
+  {
+    //return libmesh_make_unique<GradientMeshFunction>(dynamic_cast<MeshFunction *>(mesh_function.get()));
+    return libmesh_make_unique<GradientMeshFunction>(mesh_function.get());
```

Comment:
... I'm not sure how I missed this before.  Are you "cloning" here, not by making a new copy, but by giving two different unique_ptr objects control over the same raw pointer???

Reply:
And the answer is "no, you're not" - that mesh_function.get() isn't a ```GradientMeshFunction*``` that's about to get owned by a second non-unique unique_ptr, it's a ```MeshFunction*``` that's getting used as the constructor argument to a brand new unique ```GradientMeshFunction```.  Sorry about the mix up.  Apparently "how I missed this before" is "I was paying closer attention before".